### PR TITLE
Add tmrmn/CRTouch-shell

### PR DESCRIPTION
**Pull request by dennistimmermann**
* [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?

Added tmrmn/CRTouch-shell - a replacement housing for the Creality CR Touch to fit the Voron X-carriage

#### How was it tested? 

So far about 50h print time on my printer

#### Screenshots (if appropriate)

CRTouchBracket_v10_rim4

https://www.youtube.com/watch?v=5Wmlju45Rt4

### CODE REVIEW

File: printer_mods/tmrmn/CRTouch-shell/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+At the time of writing the Calibarte-Z Klipper plugin doesn't work with the BLTouch/CR Touch.
+But all thats needed is a small macro added to the `PRINT_START` routine.
+This `Z_AUTOCALIBRATE` macro probes the endstop and calculates the Z-offset from nozzle to probe.
+It then probes the bed and sets the new Z0 to the nozzle offset with a G92 before printing.
+
+
+The parameter `variable_switch_pretravel` has to be adjusted once. This is the amount the endstop switch travels before being triggered. Typically around 0.3 to 0.5mm. This is also the value to tune the prefect 1st layer.
+Additionally the exact positions of the z-endstop and middle of the bed have to be provided.
+Please read the comments in `macro-z-autocalibarte.cfg` for more information.
```
Comment (bdbell) [inline printer_mods/tmrmn/CRTouch-shell/README.md:63]: That filename looks like a typo.

File: printer_mods/tmrmn/CRTouch-shell/printer.cfg [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+speed: 5.0
+samples: 3
+samples_result: average
+sample_retract_dist: 0.5
+samples_tolerance: 0.02
+samples_tolerance_retries: 5
+pin_up_touch_mode_reports_triggered: False
+probe_with_touch_mode: True
+stow_on_each_sample: False
+pin_move_time: 0.9
```
Comment (bdbell) [inline printer_mods/tmrmn/CRTouch-shell/printer.cfg:16]: This needs a newline at the end of the file.

**Comment by bdbell** [comment]
The configuration example should not be named printer.cfg as that could be very confusing for newcomers. Please give it a different name, perhaps the name of your mod?

**Comment by bdbell** [comment]
This bit ends up being an unsupported perimeter.  I'd recommend making the chamfer bigger to extend all the way to the perimeter and making a 0.2 deep step forced bridge for printability.

image

**Comment by bdbell** [comment]
Otherwise it's a really great idea and execution, I hope the fixes can get done so we can put it in.

**Comment by dennistimmermann** [comment]
Could you maybe go into detail about the forced bridge? I could add this step, covering 50% of the hole. This would leave only straight lines when bridging. The excess perimeters would just break away when installing the screw.

bridge4

crtouch-bridging

**Comment by bdbell** [comment]
That's actually really close.  All you would need is to make that extra piece 0.8mm (2 lines) wide.  Then it would create a bridge to support the outer edge of the circle and that would be sufficient.

**Comment by newphreak** [comment]
@dennistimmermann still want this merged? or do you want to close it?
If you want it merged you also need to include a .step or .scad file in addition to the comment above from bdbell.

**Comment by cbc02009** [comment]
Closing due to inactivity.

**Pull request closed without merging**